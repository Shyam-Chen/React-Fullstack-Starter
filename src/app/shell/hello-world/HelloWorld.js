// @flow

import React from 'react';
import { compose } from 'recompose';

export const HelloWorld = (): React$Element<*> => (
  <div id="hello-world">
    <h1>Hello, World!</h1>
  </div>
);

export default compose()(HelloWorld);
