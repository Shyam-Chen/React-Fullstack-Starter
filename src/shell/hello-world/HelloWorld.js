import React, { useState } from 'react';

import { dynamic } from '~/core/store';

// import * as actions from './actions';
// import * as selectors from './selectors';
import reducer from './reducer';

const HelloWorld = () => {
  const [word] = useState('World');

  return (
    <div id="hello-world">
      <h1>Hello, {word}!</h1>
    </div>
  );
};

export default dynamic('helloWorld', reducer)(HelloWorld);
