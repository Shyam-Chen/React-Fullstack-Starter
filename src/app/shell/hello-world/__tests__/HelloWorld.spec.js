import React from 'react';
import { shallow } from 'enzyme';

import { HelloWorld } from '../HelloWorld';

describe('HelloWorld', () => {
  it('should render an initial component', () => {
    const component = shallow(
      <HelloWorld />,
    );

    expect(component).toMatchSnapshot();
  });
});
