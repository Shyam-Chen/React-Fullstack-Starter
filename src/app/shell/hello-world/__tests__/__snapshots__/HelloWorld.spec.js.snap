// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld should render an initial component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HelloWorld />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        Hello, World!
      </h1>,
      "id": "hello-world",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Hello, World!",
      },
      "ref": null,
      "rendered": "Hello, World!",
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          Hello, World!
        </h1>,
        "id": "hello-world",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Hello, World!",
        },
        "ref": null,
        "rendered": "Hello, World!",
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
